jazz_create-serverless-service/test/test.js

```diff
@@ -22,234 +22,667 @@ const awsContext = require('aws-lambda-mock-context');
+      });
 
-    /*
-    * Given successful parameters and setup, handler() should send a POST http request
-    * @params {object, object, function} default event, aws context, callback
-    * @returns index.handler() should attempt an http POST if given valid paramters
-    */
-    it("should send an http POST given valid input parameters", function(){
-      //wrapping the Request() method that gets internally called by node request.js for any http method
+      index.getToken(config).then(() => {
+        assert.fail()
+      }).catch((err) => {
+        if (err.error && err.error === errMessage) {
+          bool = true
+        }
+        assert.isTrue(bool);
+      })
+      reqStub.restore();
+    })
+
+  })
+  describe("getServiceData", () => {
+    beforeEach(function () {
+      event = {
+        "stage": "test",
+        "headers": {
+          "Authorization": "fr1end$hip_1s_mAg1c"
+        },
+        "principalId": "@pp1eJack",
+        "body": {
+          "service_name": "test-service",
+          "service_type": "function",
+          "domain": "test-domain",
+          "runtime": "nodejs",
+          "approvers": ['tw1light_$pArkle'],
+          "rateExpression": "1 * * * ? *",
+          "slack_channel": "mlp_fim",
+          "require_internal_access": false,
+          "create_cloudfront_url": false //, //?
+          //"enableEventSchedule" : false
+        }
+      };
+      service_creation_data = event.body;
+      context = awsContext();
+      callback = (err, responseObj) => {
+        if (err) {
+          return err;
+        } else {
+          return JSON.stringify(responseObj);
+        }
+      };
+    });
+    it("should should return error when passed an invalid rateExpression", () => {
+      let cronValues = [null, "", "P!nk!e_P!e"];
+      let authToken = "temp-auth-token";
+      let config = configObj(event);
+      let bool = false;
+      for (let cron in cronValues) {
+        service_creation_data.rateExpression = cron
+        index.getServiceData(service_creation_data, authToken, config).then().catch((errorMsg) => {
+
+          if (errorMsg.result === "invalid") {
+            bool = true;
+          }
+          assert.isTrue(bool);
+        })
+      }
+
+    });
+    it("should return input object with METADATA values for valid input parameters for service type function (event_source e2c)", () => {
+      let authToken = "temp-auth-token";
+      let bool = false;
+      service_creation_data["enableEventSchedule"] = true;
+      service_creation_data.event_source_ec2 = "temp-url";
+      service_creation_data.event_action_ec2 = "temp-url";
+      service_creation_data.event_source_ec2 = "sample-test-data"
+      let config = configObj(event);
+      index.getServiceData(service_creation_data, authToken, config).then((input) => {
+        if (input.METADATA.eventScheduleEnable && input.METADATA.eventScheduleRate != null && input.METADATA.eventScheduleRate != "") {
```

Comment:
expectation should be to set correctly that you can assert and not have to do 'if' here (similar)

Reply:
the expectation is that metadata values should be populated by the function 
if input.METADATA values are not populated , the test case fails